src/main/java/dev/snowdrop/release/services/GitService.java

```diff
@@ -191,23 +216,60 @@ private GitConfig(String org, String repo, String branch) {
+         * with the specified token and optionally cloning from the specified branch if the specified {@code gitRef} points to a
+         * non-existing branch.
+         *
+         * @param gitRef          a String in the {@code organization/repository/branch} format where {@code branch} can point
+         *                        to a non-existing branch
+         * @param token           the GitHub token to use for the operations
+         * @param cloneFromBranch the name of the branch to clone from if the branch specified by {@code gitRef} doesn't exist.
+         *                        If {@code null} is specified, then {@link #DEFAULT_CLONE_FROM_BRANCH} will be used if the
+         *                        desired branch needs to be created
+         * @return the {@link GitConfig} needed to operate on the repository
+         */
+        public static GitConfig githubConfig(String gitRef, String token, String cloneFromBranch) {
             final var split = gitRef.split("/");
             if (split.length != 3) {
                 throw new IllegalArgumentException("Invalid git reference: " + gitRef
                     + ". Must follow organization/repository/branch format.");
             }
-            return new GitHubConfig(split[0], split[1], split[2], token);
+            return new GitHubConfig(split[0], split[1], split[2], token, cloneFromBranch);
         }
 
-        public static GitConfig gitlabConfig(String gitRef, String release, String username, String token) {
+        /**
+         * Creates a {@link GitConfig} adapted to operate on the GitLab repository identified by the specified parameters,
+         * creating the associated branch if needed.
+         *
+         * @param release         the Spring Boot release
+         * @param username        the GitLab user name
+         * @param token           the GitLab token
+         * @param gitRef          the git reference identifying the repository in the {@code organization/repository} format
+         * @param cloneFromGitRef the name of the branch to clone from if the branch specified by {@code gitRef} doesn't exist.
+         *                        If {@code null} is specified, then {@link #DEFAULT_CLONE_FROM_BRANCH} will be used if the
+         *                        desired branch needs to be created
+         * @return the {@link GitConfig} needed to operate on the repository
+         */
+        public static GitConfig gitlabConfig(String release, String username, String token, String gitRef, String cloneFromGitRef) {
             final var split = gitRef.split("/");
             if (split.length != 2) {
                 throw new IllegalArgumentException("Invalid git reference: " + gitRef
                     + ". Must follow organization/repository format.");
             }
             final var branch = "snowdrop-issues-manager-" + release;
```

Comment:
We should probably validate the release format here?

Reply:
Implemented - https://github.com/jacobdotcosta/release-manager/blob/init-release-folders/src/main/java/dev/snowdrop/release/services/GitService.java#L260